caldav/objects.py

```diff
@@ -811,7 +811,7 @@ def object_by_uid(self, uid, comp_filter=None):
             # attempting a match.
             item_uid = re.search(r'\nUID:((.|\n[ \t])*)\n', item.data)
             if (not item_uid or
-                    re.sub(r'\n[ \t]', '', item_uid.group(1)) != uid):
+                    re.sub(r'\n[ \t]', '', item_uid.group(1).rstrip()) != uid):
```

Comment:
I go this while debugging:
```
(Pdb) item_uid
<re.Match object; span=(334, 373), match='\nUID:19970901T130000Z-123405@host.com\r\n'>
(Pdb) item_uid.group(1)
'19970901T130000Z-123405@host.com\r'
```
It looks like there is a problem in the regex when parsing Windows-style line endings. The ID returned from `Todo.vobject_instance.vtodo` is `"19970901T130000Z-123405@host.com"`, but here the extra `\r` character means the IDs do not match. This change didn't seem to introduce any errors in the rest of the tests, but I don't have enough context to know if this will cause issues elsewhere (obviously it's a pretty ham-fisted fix, anyway). I would want to add some unit tests for this method before comitting a change here.

Reply:
good catch

Reply:
There was also another issue report about breakages due to windows line endings, so it I've simply stripped out all instances of `\r` in `DAVClient.DAVResponse.__init__`, though since this is icalendar-specific, it should probably be put into another layer of the code.